// node-test-wrapper/lib/inquirer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _inquirer = _interopRequireDefault(require("inquirer"));

var _message = require("./utils/message");

var _file = require("./utils/file");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var questions = [{
  name: 'esVersion',
  type: 'list',
  message: 'Which version of javascript do you want to generate?',
  choices: ['es5', 'es6']
}, {
  name: 'authOpt',
  type: 'list',
  message: 'Which type of authentication does your app use?',
  choices: ['none', 'basic', 'token', 'session']
}, {
  name: 'filePath',
  type: 'input',
  "default": 'test/testUtils/app.js',
  message: 'Specify file path for the generated supertest wrapper?',
  filter: function filter(input) {
    return input.trim();
  },
  validate: function validate(input) {
    if (!input) {
      return false;
    }

    var available = (0, _file.fileAvailable)(input);
    return available === true ? available : (0, _message.toWarning)(available);
  }
}];
/**
 * @returns {Promise}
 */

var inquire = function inquire() {
  return _inquirer["default"].prompt(questions);
};

var _default = inquire;
exports["default"] = _default;
